### ISSUE TITLE
Missing test for accept header audio

### ISSUE THREAD

**Issue by StephenHodgson** [huggingface/huggingface_hub#1834]
We're testing the image header accept requests for valid return object encoding but not any of the audio tasks.

https://github.com/huggingface/huggingface_hub/blob/ef846d759f0aee9d586ecdffaa09c49ae0807e9c/tests/test_inference_client.py#L512

Related to https://github.com/huggingface/api-inference-community/issues/351

suggested test casees:

- `audio/mp3`
- `audio/wav`

**Comment by Wauplin** [comment]
> We're testing the image header accept requests for valid return object encoding but not any of the audio tasks.

Not exactly. We are testing that if a task expects an image to be returned (i.e. "text-to-image" or"image-to-image"), then the `Accept` header must be set to png by default in the requests (more details in https://github.com/huggingface/huggingface_hub/pull/1567).
But we are not testing the response from the Inference API. In general, I don't think we should use `huggingface_hub` test pipeline to test the Inference API (it's too compute intensive + tests are mocked anyway + InferenceClient is meant to work on [several backends](https://github.com/huggingface/huggingface_hub/pull/1567#issuecomment-1651272098)).

_If_ we want to test that, it should be in https://github.com/huggingface/api-inference-community but I doubt it to be honest.

**Comment by Wauplin** [comment]
kinda stale / revamped with inference providers
